vmdb/app/models/mixins/container_provider_mixin.rb

```diff
@@ -42,11 +43,12 @@ def api_endpoint
   end
 
   def connect(options = {})
-    hostname = options[:hostname] || address
-    port     = options[:port] || self.port
-    username = options[:user] || authentication_userid(options[:auth_type])
-    password = options[:pass] || authentication_password(options[:auth_type])
-    self.class.raw_connect(hostname, port, username, password, options[:service])
+    options[:hostname] ||= address
+    options[:port] ||= self.port
```

Comment:
Should hostname and port be removed from options before passing the hash to `raw_connect` method?